Swerve2022-Imported/src/main/java/frc/robot/Robot.java

```diff
@@ -46,7 +46,7 @@ public void robotInit( )
 
     // Instantiate our RobotContainer. This will perform all our button bindings, and put our autonomous
     // chooser on the dashboard.
-    m_robotContainer = RobotContainer.getInstance( );
+    m_robotContainer = new RobotContainer( );

```

Comment:
why does replacing the .getInstance() method with creating a new command help fix the pointer error?

Reply:
We should go over this in person, but the sequence of robotInit creating the robotContainer object which then created subsystems was ok, but some of the subsystems were then creating references to other subsystems that did not exist yet. It was all order dependent (and still is a bit). This is not good form, but a challenge to fix correctly right now, so I tried to make it less obscure in the code.

Reply:
BTW, there are technically no "pointers" in java. It's called a "reference" to an object, since it is creating the whole object in m_robotContainer (and not just returning a pointer to an object). So we've actually created the object in the Robot.java file here. In the prior version, the object was created in robotContainer.java and a reference was returned.

Reply:
ok, that makes sense.  You mentioned that this is a quick fix, so what would the official way to fix this be?

Reply:
The right way is to not access subsystems from a different subsystem (we access vision and LED from shooter, etc.). I don't have a straightforward solution for that right now without writing extra commands to get around it. As long as we don't muck with the ordering of how the subsystems are instantiated, it will be fine. Can look at a better way to do it later.